tests/functional/afr/test_self_heal_with_full_algorithm.py

```diff
@@ -0,0 +1,207 @@
+            brick_to_bring_offline))
+
+        # Create few files under the directory with data
+        cmd = ("cd %s/test_full_self_heal ; for i in `seq 1 100` ; "
+               "do dd if=/dev/urandom of=file.$i bs=1M count=1; done;"
+               % (mountpoint))
+        ret, _, _ = g.run(client, cmd)
+        self.assertEqual(ret, 0, "Failed to create file on mountpoint")
+        g.log.info("Successfully created files on mountpoint")
+
+        # Start volume with force to bring all bricks online
+        ret, _, _ = volume_start(self.mnode, self.volname, force=True)
+        self.assertEqual(ret, 0, "Volume start with force failed")
+        g.log.info("Volume: %s started successfully", self.volname)
+
+        # Verify volume's all process are online
+        ret = verify_all_process_of_volume_are_online(self.mnode, self.volname)
+        self.assertTrue(ret, ("Volume %s : All process are not online",
+                              self.volname))
+
+        # Monitor heal completion
+        self.assertTrue(monitor_heal_completion(self.mnode, self.volname,
+                                                interval_check=10),
+                        "Heal failed after 20 mins")
+
+        # Check are there any files in split-brain
+        self.assertFalse(is_volume_in_split_brain(self.mnode, self.volname),
+                         "Some files are in split brain for "
+                         "volume: {}".format(self.volname))
+
+        # Check arequal checksum of all the bricks are same
+        for subvol in subvols:
+            ret, arequal_from_the_bricks = collect_bricks_arequal(subvol)
+            self.assertTrue(ret, "Arequal is collected successfully across "
+                                 "the bricks in the subvol {}".format(subvol))
+            cmd = len(set(arequal_from_the_bricks))
+            if (self.volume_type == "arbiter" or
+                    self.volume_type == "distributed-arbiter"):
+                cmd = len(set(arequal_from_the_bricks[:2]))
+            self.assertEqual(cmd, 1, "Arequal"
+                             " is same on all the bricks in the subvol")
+
+        # List another brick in each subvol and bring them offline
+        brick_to_bring_offline = []
+        for subvol in subvols:
+            self.assertTrue(subvol, "List is empty")
+            brick_to_bring_offline.extend(sample(subvol, 2))
+
+        ret = bring_bricks_offline(self.volname, brick_to_bring_offline)
+        self.assertTrue(ret, "Unable to bring brick: {} offline".format(
+            brick_to_bring_offline))
+
+        # Validate the brick is offline
+        ret = are_bricks_offline(self.mnode, self.volname,
+                                 brick_to_bring_offline)
+        self.assertTrue(ret, "Brick:{} is still online".format(
+            brick_to_bring_offline))
+
+        # Modify files under test_full_self_heal directory
+        cmd = ("for i in `seq 1 100` ; do truncate -s 0 file.$i ; "
```

Comment:
cmd and above comment is differing, please `cd` into created dir.

Reply:
We have already cd'ed into the directory with the first write operation and did not come out of it.

Reply:
requires fix

Reply:
Done.